c10/cuda/CUDAStream.cpp

```diff
@@ -18,9 +18,19 @@ namespace {
 // Internal implementation is entirely hidden
 struct CUDAStreamInternals {
   CUDAStreamInternals() = default;
+  CUDAStreamInternals(const CUDAStreamInternals&) = delete;
```

Comment:
nit: should we use a macro for this to avoid potentially missing ctors/assignments?